src/types/children.js

```diff
@@ -3,33 +3,55 @@ import get from "lodash.get";
 import assign from "lodash.assign";
 
 import input from "./lib/input.js";
+import checkHidden from "./lib/hide.js";
+import addClasses from "./lib/classer.js";
 
 import css from "./lib/types.css";
 
 // Bound below
 var types;
 
 export function view(ctrl, options) {
-    var fields = options.fields || [];
+    var fields = options.fields || [],
+        registerHidden = options.registerHidden,
+        mFields = [];
 
-    return m("div", options.class ? { class : options.class } : null,
-        fields.map(function(field, index) {
-            var component = types[field.type || field];
-
-            if(!component) {
-                return m("div",
-                    m("p", "Unknown component"),
-                    m("pre", JSON.stringify(field, null, 4))
-                );
+    mFields = fields.map(function(field, index) {
+        var component,
+            wasHidden,
+            result;
+
+        component = types[field.type || field];
+
+        if(field.show) {
+            wasHidden = field.show.hidden;
+            field.show.hidden = checkHidden(options.state, field);
+
+            if(registerHidden && field.show.hidden !== wasHidden) {
+                // hidden status changed, notify the controller.
+                registerHidden(field.key, field.show.hidden);
             }
+        }
 
-            return m.component(component, assign({}, options, {
-                field : field,
-                class : css[index ? "field" : "first"],
-                data  : get(options.data, field.key),
-                path  : options.path.concat(field.key)
-            }));
-        })
+        if(!component) {
```

Comment:
This check seems like it should go above any field visibility bits. No reason to even bother if it's an unrecognized component type.